flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/TaskExecutorPartitionTrackerImpl.java

```diff
@@ -42,12 +42,12 @@ public TaskExecutorPartitionTrackerImpl(ShuffleEnvironment<?, ?> shuffleEnvironm
 	}
 
 	@Override
-	public void startTrackingPartition(JobID producingJobId, ResultPartitionID resultPartitionId, IntermediateDataSetID intermediateDataSetId) {
+	public void startTrackingPartition(JobID producingJobId, ResultPartitionID resultPartitionId, IntermediateDataSetID intermediateDataSetId, int numberOfPartitions) {
```

Comment:
it could be then just:
`startTrackingPartition(JobID, TaskExecutorPartitionInfo/ResultPartitionDeploymentDescriptor)`
like in JM.
`TaskExecutorPartitionInfo` could have a builder or factory method.
Maybe we could even track RPDD instead of `TaskExecutorPartitionInfo` if it is enough in future.